lib/dialects/postgres/query-generator.js

```diff
@@ -102,8 +102,10 @@ var QueryGenerator = {
     });
   },
 
-  showTablesQuery: function() {
-    return "SELECT table_name FROM information_schema.tables WHERE table_schema = 'public' AND table_type LIKE '%TABLE' AND table_name != 'spatial_ref_sys';";
+  showTablesQuery: function(options) {
+    options = options || {};
+    options.schema = options.schema || 'public';
```

Comment:
Right, but take a look at the extractTableDetails method since that might help you here.